Replace delimited

dkubb opened the pull request:
This branch changes the s-expressions to use names that are better at describing the node that "delimited".

dkubb committed:
Change s-expressions to be better named

dkubb commented:
@mbj I wanted to show you what I was talking about earlier to see how you felt about the change.

mbj commented:
@dkubb I agree with `delimited` is a bad name. Maye we should just use `tuple`, or anything else that is generic enough to use as "this is a list of specifiers". Maybe just `list`. I dont think we should add unique-by-name nodes like `where` that can _only_ occur at a specific children index of a specific parent node. So I'm totally okay with renaming delemited to another name, but IMHO we should NOT split it up.

The position already tells the role of a child node, no need to introduce a child node that is only used once in the specific position.

dkubb commented:
It just looks so ugly when I stringify the s-expression when debugging. The ugliness is making me question whether I should trust my gut and give it a good name.

dkubb committed:
Rename select identifier as from

* The from part of a SELECT query is not necessarily always an
  identifier. It's an expression that evaluates to a table value,
  which can be a single identifier, but it can also be a subquery,
  or a join expression.

dkubb commented:
I'm going to merge this in with a few small chages, if it turns out to the be the wrong decision changing things back will be relatively trivial.

mbj commented:
> It just looks so ugly when I stringify the s-expression when debugging. The ugliness is making me question whether I should trust my gut and give it a good name.

This is because SQL is an ugly language ;)
